pkg/gardenlet/controller/seed/seed_care_control.go

```diff
@@ -0,0 +1,221 @@
+func NewCareReconciler(
+	clientMap clientmap.ClientMap,
+	l logrus.FieldLogger,
+	config *config.GardenletConfiguration,
+) reconcile.Reconciler {
+	return &careReconciler{
+		clientMap: clientMap,
+		logger:    l,
+		config:    config,
+	}
+}
+
+type careReconciler struct {
+	logger    logrus.FieldLogger
+	clientMap clientmap.ClientMap
+	config    *config.GardenletConfiguration
+}
+
+func (r *careReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	log := r.logger.WithField("seed", req.String())
+
+	gardenClient, err := r.clientMap.GetClient(ctx, keys.ForGarden())
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get garden client: %w", err)
+	}
+
+	seed := &gardencorev1beta1.Seed{}
+	if err := gardenClient.Client().Get(ctx, req.NamespacedName, seed); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Infof("[SEED CARE] Stopping care operations for Seed since it has been deleted")
+			return reconcile.Result{}, nil
+		}
+		log.Infof("[SEED CARE] unable to retrieve object from store: %+v", err)
+		return reconcile.Result{}, err
+	}
+
+	if err := r.care(ctx, gardenClient, seed, log); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{RequeueAfter: r.config.Controllers.ShootCare.SyncPeriod.Duration}, nil
+}
+
+var (
+	// NewSeed is used to create a new `operation.Operation` instance.
+	NewSeed = defaultNewSeedObjectFunc
+	// NewHealthCheck is used to create a new Health check instance.
+	NewHealthCheck = defaultNewHealthCheck
+)
+
+func (r *careReconciler) care(ctx context.Context, gardenClientSet kubernetes.Interface, seed *gardencorev1beta1.Seed, log logrus.FieldLogger) error {
+	careCtx, cancel := context.WithTimeout(ctx, r.config.Controllers.ShootCare.SyncPeriod.Duration)
+	defer cancel()
+
+	log.Infof("[SEED CARE] Starting care")
+	gardenClient := gardenClientSet.Client()
+
+	// Initialize conditions based on the current status.
+	conditionTypes := []gardencorev1beta1.ConditionType{
+		gardencorev1beta1.SeedSystemComponentsHealthy,
+	}
+	var conditions []gardencorev1beta1.Condition
+	for _, cond := range conditionTypes {
+		conditions = append(conditions, gardencorev1beta1helper.GetOrInitCondition(seed.Status.Conditions, cond))
+	}
+
+	seedClient, err := r.clientMap.GetClient(careCtx, keys.ForSeed(seed))
+	if err != nil {
+		log.Errorf("seedClient cannot be constructed: %s", err.Error())
+
+		if err := careSetupFailure(ctx, gardenClient, seed, "Precondition failed: seed client cannot be constructed", conditions); err != nil {
+			log.Error(err)
+		}
+
+		return nil // We do not want to run in the exponential backoff for the condition checks.
+	}
+
+	seedObj, err := NewSeed(ctx, seed)
+	if err != nil {
+		log.Errorf("seedObj cannot be constructed: %s", err.Error())
+		return nil
+	}
+
+	staleExtensionHealthCheckThreshold := confighelper.StaleExtensionHealthChecksThreshold(r.config.Controllers.ShootCare.StaleExtensionHealthChecks)
```

Comment:
Hm, why is this needed. The doc string says that it's about checking extension CRDs and this does not happen in this controller, right?

https://github.com/gardener/gardener/blob/179cf336ce5cfaa8dc3b0ef08ea1aba87ee0765c/pkg/gardenlet/apis/config/types.go#L286-L294